packages/builders/src/interactions/commands/chatInput/ChatInputCommandSubcommands.ts

```diff
@@ -0,0 +1,103 @@
+import type { JSONEncodable } from '@discordjs/util';
+import type {
+	APIApplicationCommandSubcommandOption,
+	APIApplicationCommandSubcommandGroupOption,
+} from 'discord-api-types/v10';
+import { ApplicationCommandOptionType } from 'discord-api-types/v10';
+import { Mixin } from 'ts-mixer';
+import { isValidationEnabled } from '../../../util/validation.js';
+import { chatInputCommandSubcommandGroupPredicate, chatInputCommandSubcommandPredicate } from './Assertions.js';
+import { SharedChatInputCommandOptions } from './mixins/SharedChatInputCommandOptions.js';
+import type { SharedNameAndDescriptionData } from './mixins/SharedNameAndDescription.js';
+import { SharedNameAndDescription } from './mixins/SharedNameAndDescription.js';
+
+export interface ChatInputCommandSubcommandGroupData {
+	options?: ChatInputCommandSubcommandBuilder[];
+}
+
+/**
+ * Represents a folder for subcommands.
+ *
+ * @see {@link https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups}
+ */
+export class ChatInputCommandSubcommandGroupBuilder
+	extends SharedNameAndDescription
+	implements JSONEncodable<APIApplicationCommandSubcommandGroupOption>
+{
+	protected declare readonly data: ChatInputCommandSubcommandGroupData & SharedNameAndDescriptionData;
+
+	/**
+	 * Adds a new subcommand to this group.
+	 *
+	 * @param input - A function that returns a subcommand builder or an already built builder
+	 */
+	public addSubcommand(
+		input:
+			| ChatInputCommandSubcommandBuilder
+			| ((subcommandGroup: ChatInputCommandSubcommandBuilder) => ChatInputCommandSubcommandBuilder),
+	) {
+		// eslint-disable-next-line @typescript-eslint/no-use-before-define
+		const result = typeof input === 'function' ? input(new ChatInputCommandSubcommandBuilder()) : input;
+
+		this.data.options ??= [];
+		this.data.options.push(result);
+
+		return this;
+	}
+
+	/**
+	 * Serializes this builder to API-compatible JSON data.
+	 *
+	 * Note that by disabling validation, there is no guarantee that the resulting object will be valid.
+	 *
+	 * @param validationOverride - Force validation to run/not run regardless of your global preference
+	 */
+	public toJSON(validationOverride?: boolean): APIApplicationCommandSubcommandGroupOption {
+		const { options, ...rest } = this.data;
+
+		const data = {
+			type: ApplicationCommandOptionType.SubcommandGroup as const,
+			...(structuredClone(rest) as Omit<APIApplicationCommandSubcommandGroupOption, 'type'>),
```

Comment:
Consistency with `ChatInputCommandSubcommandBuilder#toJSON()`

```suggestion
			...(structuredClone(rest) as Omit<APIApplicationCommandSubcommandGroupOption, 'type'>),
			type: ApplicationCommandOptionType.SubcommandGroup as const,
```

Reply:
@didinele you marked this as resolved but didn't resolve it